src/rgw/rgw_lc.cc

```diff
@@ -1441,52 +1475,12 @@ class LCOpAction_Transition : public LCOpAction {
             null_yield);
     auto version_id = oc.o.key.instance;
 
-    ret = notify->publish_reserve(oc.dpp, nullptr);
-    if (ret < 0) {
+    r = notify->publish_reserve(oc.dpp, nullptr);
```

Comment:
the event_types above doesn't take `!(bucket->versioned)` case into account. Can you please address that too ..

if (! oc.bucket->versioned()) {
     event_types.push_back(rgw::notify::ObjectTransition);
}  else if (oc.o.is_current() && !oc.o.is_delete_marker()) {
      event_types.insert(event_types.end(),
                         {rgw::notify::ObjectTransitionCurrent,
                          rgw::notify::LifecycleTransition});
    } else {
      event_types.push_back(rgw::notify::ObjectTransitionNoncurrent);
    }

Reply:
sure.

Reply:
I'm not sure LifecycleTransition is working correctly, either

Reply:
`    rgw::notify::EventTypeList event_types;
    event_types.insert(event_types.end(), {rgw::notify::LifecycleTransition});
    if ((!bucket->versioned()) ||
	(bucket->versioned() && oc.o.is_current() && !oc.o.is_delete_marker())) {
      event_types.push_back(rgw::notify::ObjectTransitionCurrent);
    } else {
      event_types.push_back(rgw::notify::ObjectTransitionNoncurrent);
    }`

Reply:
how about:
`    rgw::notify::EventTypeList event_types;
    event_types.insert(event_types.end(), {rgw::notify::LifecycleTransition});
    if ((!bucket->versioned()) ||
	(bucket->versioned() && oc.o.is_current() && !oc.o.is_delete_marker())) {
      event_types.push_back(rgw::notify::ObjectTransitionCurrent);
    } else {
      event_types.push_back(rgw::notify::ObjectTransitionNoncurrent);
    }`